vision/fiducial_detection_and_pose.hh

```diff
@@ -30,7 +30,18 @@ struct MarkerInWorld {
 
 std::vector<MarkerDetection> detect_markers(const cv::Mat& mat);
 
-std::vector<MarkerInWorld> get_world_from_marker_centers(
-    const cv::Mat& camera_image, const SE3& world_from_camera);
+std::vector<MarkerInWorld> get_world_from_marker_centers(const cv::Mat& camera_image,
+                                                         const SE3& world_from_camera);
+
+std::optional<SE3> detect_board(const cv::Mat& input_image);
+
+const cv::Ptr<cv::aruco::Dictionary> aruco_dictionary =
```

Comment:
Make a function that returns this instead of setting it as a global constant.

The reason for this is to avoid [The Static Initialization Order Fiasco](https://isocpp.org/wiki/faq/ctors)

Reply:
(Do the same for the aruco_board constant as well)